runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/fn/control/BeamFnMapTaskExecutor.java

```diff
@@ -139,30 +150,44 @@ private AutoCloseable progressTrackerCloseable(ProgressTracker progressTracker)
    */
   @Override
   public Iterable<CounterUpdate> extractMetricUpdates() {
+    List<CounterUpdate> result = progressTracker.extractCounterUpdates();
+    if ((result != null) && (result.size() > 0)) {
+      return result;
+    }
+
+    // BeamFnApi.Metrics was deprecated and replaced with more flexible MonitoringInfo.
+    // However some of SDKs have implementations that utilize BeamFnApi.Metrics and were not yet
+    // migrated to using new approach.
+    // Falling back to using deprecated approach until all officially supported SDKs complete migration.
```

Comment:
Is there a JIRA that tracks this migration?

More generally: is it possible to make this comment more actionable? Something of the form: `once this condition is true (all SDKs are migrated), do this action (remove this fallback)`. It should be straightforward for anybody to check on the migration status and then delete the code.